### PR TITLE
FIX: JsonParser / ParseStringValue now creates unescaped strings

### PR DESCRIPTION
**Pull request by AlexTyrer**
### Description

Comparing escaped and non-escaped strings is very slow - this happens when devices are deactivated/reactivated such as when the Quest2 headset and controllers enter and awake from sleep mode.

The high cost comes from having to compare each string character-by-character (skipping '\' escapes) each time entering the C# runtime to perform System.Globalization.ToUpperInvariant.

### Changes made

This change to TryMatchDisconnectedDevice() means that any parsed strings that contain escape characters are converted to non-escaped strings for much quicker comparison.

This saves approximately 0.5ms / frame when an Oculus headset or controller is reactivated.

### Testing

Local testing with the Quest2 headset and controllers using Superluminal to capture the cost of device awakening.

### Risk

We allocate a temporary string where before we did not.

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [x] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [x] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [x] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [x] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Utilities/PrimitiveValue.cs [hunk 1]
```diff
@@ -450,7 +450,7 @@ public bool ToBoolean(IFormatProvider provider = null)
                 case TypeCode.Single:
                     return !Mathf.Approximately(m_FloatValue, default);
                 case TypeCode.Double:
-                    return NumberHelpers.Approximately(m_DoubleValue, default);
+                    return !NumberHelpers.Approximately(m_DoubleValue, default);
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Utilities/PrimitiveValue.cs:453]: Bonus bug fix?

**Review by ekcoh — APPROVED** [review]
LGTM, looks like a good improvement @AlexTyrer

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ however, it has to be formatted properly to pass verification tests.
 
 ### Changed
 - Use `ProfilerMarker` instead of `Profiler.BeginSample` and `Profiler.EndSample` when appropriate to enable recording of profiling data.
+- Improved performance of disconnected device activation (ISX-1450)
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: I think this could live under ### Fixed since it's not really a changed behavior, right?

**Review by ekcoh — APPROVED** [review]

File: Packages/com.unity.inputsystem/InputSystem/Utilities/PrimitiveValue.cs (hunk 1 above)
Comment (AlexTyrer) [reply Packages/com.unity.inputsystem/InputSystem/Utilities/PrimitiveValue.cs:453]: Yes, I spotted a bug we didn't even know we had!

**Pull request merged**